pydeseq2/dds.py

```diff
@@ -593,22 +569,39 @@ def calculate_cooks(self):
         """
 
         # Check that MAP dispersions are available. If not, compute them.
-        if not hasattr(self, "dispersions"):
+        if "dispersions" not in self.varm:
             self.fit_MAP_dispersions()
 
-        num_vars = self.design_matrix.shape[1]
+        num_vars = self.obsm["design_matrix"].shape[-1]
+        nz_data = self[:, self.non_zero_genes]
```

Comment:
maybe call it `nonzero_data` rather than `nz_data`? more explicit name (for 2s I hesitated with New Zealand...)